pkg/resource/clickpipe.go

```diff
@@ -1094,6 +1100,83 @@ func (c *ClickPipeResource) ModifyPlan(ctx context.Context, request resource.Mod
 		plan.TriggerResync = types.BoolValue(false)
 		response.Diagnostics.Append(response.Plan.Set(ctx, plan)...)
 	}
+
+	// Validate Postgres table mappings changes
+	if !request.State.Raw.IsNull() {
+		var planSourceModel, stateSourceModel models.ClickPipeSourceModel
+		if diags := plan.Source.As(ctx, &planSourceModel, basetypes.ObjectAsOptions{}); !diags.HasError() {
+			if diags := state.Source.As(ctx, &stateSourceModel, basetypes.ObjectAsOptions{}); !diags.HasError() {
+				// Only validate if this is a Postgres source
+				if !planSourceModel.Postgres.IsNull() && !stateSourceModel.Postgres.IsNull() {
+					var planPostgres, statePostgres models.ClickPipePostgresSourceModel
+					if diags := planSourceModel.Postgres.As(ctx, &planPostgres, basetypes.ObjectAsOptions{}); !diags.HasError() {
+						if diags := stateSourceModel.Postgres.As(ctx, &statePostgres, basetypes.ObjectAsOptions{}); !diags.HasError() {
+
+							// Get state and plan table mappings
+							var stateMappings, planMappings []models.ClickPipePostgresTableMappingModel
+							if !statePostgres.TableMappings.IsNull() {
+								stateMappings = make([]models.ClickPipePostgresTableMappingModel, len(statePostgres.TableMappings.Elements()))
+								statePostgres.TableMappings.ElementsAs(ctx, &stateMappings, false)
+							}
+							if !planPostgres.TableMappings.IsNull() {
+								planMappings = make([]models.ClickPipePostgresTableMappingModel, len(planPostgres.TableMappings.Elements()))
+								planPostgres.TableMappings.ElementsAs(ctx, &planMappings, false)
+							}
+
+							// Validation 1: Cannot delete the last table mapping
+							if len(stateMappings) == 1 && len(planMappings) == 0 {
```

Comment:
Are stateMappings guaranteed to go through 1 before attempting 0? Going from 2 to 0 would also be a mistake

Reply:
Yes you are correct, I think my head was wrapped in the logic of deleting/adding new mappings on the same request. We should just check that planMappings is never 0 as this will cause OpenAPI validation to fail.